flake.nix

```diff
@@ -0,0 +1,102 @@
+{
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+
+    onix = {
+      url = "github:rizo/onix/3740a2beb84bc21860385f8bbb0d5f557842de03";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs =
+    {
+      self,
+      nixpkgs,
+      flake-utils,
+      onix,
+    }@inputs:
+    flake-utils.lib.eachDefaultSystem (
+      system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        onix' = onix.packages.${system}.latest;
+        compilerVersion = "5.2.0";
+        opamFiles = [
+          ./imandra-document.opam
+        ];
+        onixEnv = onix'.env {
+          path = ./.;
+          roots = opamFiles;
+          lock = ./onix-lock.json;
+          deps = {
+            "ocaml-system" = compilerVersion;
+          };
+        };
+        onixEnvDev = onix'.env {
+          path = ./.;
+          roots = opamFiles;
+          lock = ./onix-lock-dev.json;
+          deps = {
+            "ocaml-system" = compilerVersion;
+            "ocaml-lsp-server" = "*";
+            "utop" = "*";
+          };
+        };
+        onixLock =
+          let
+            shellHook = pkgs.writeText "onix-lock-shellHook" onixEnv.lock.shellHook;
+            lockPatched = pkgs.runCommand "onix-lock-shellHook-patched" { } ''
+              sed 's:--lock-file=/nix/store/.*/onix-lock.json::' "${shellHook}" > "$out"
+              chmod +x "$out"
+            '';
+          in
+          pkgs.writeShellApplication {
+            name = "onix-lock";
+            runtimeInputs = [ onix' ];
+            text = lockPatched;
+          };
+        onixLockDev =
+          let
+            shellHook = pkgs.writeText "onix-lock-dev-shellHook" onixEnvDev.lock.shellHook;
```

Comment:
Some notes about `onix` that seem relevant to figuring out why `onix-lock-dev` behaves oddly:

`onixEnvDev.lock.shellHook` resolves to a `shellHook` defined by a `pkgs.mkShell` call in [onix/nix/core.lock](https://github.com/rizo/onix/blob/3740a2beb84bc21860385f8bbb0d5f557842de03/nix/core.nix#L331). I don't see anything unusual about the `lockPath` argument to `lock`.

Similarly, nothing seems unusual about `cmdliner` code for `lock_file_arg` in [src/onix/Main.ml](https://github.com/rizo/onix/blob/3740a2beb84bc21860385f8bbb0d5f557842de03/src/onix/Main.ml#L28).